src/splight_agent/models.py

```diff
@@ -89,13 +124,33 @@ class Component(APIObject):
     deployment_updated_at: Optional[str]
     compute_node: Optional[str]
 
+    @property
+    def instance_type(self):
+        return self.__class__.__name__.lower()
+
+    @abstractmethod
+    def get_hub_instance(self):
+        pass
+
+    @abstractmethod
+    def get_deploy_label(self):
+        pass
```

Comment:
These abstract methods are tricky, in any of the two implementations the signature is the same. For example, in one case the `get_hub_instances` returns a `HubComponent` instances meanwhile the second case returns a `HubServer`